core/src/test/java/apoc/export/cypher/ExportCypherTest.java

```diff
@@ -163,6 +159,91 @@ public void testExportQueryCypherForNeo4j() {
         assertEquals(EXPECTED_NEO4J_SHELL, readFile(fileName));
     }
 
+    @Test
+    public void testExportQueryOnlyRelCypherForNeo4j() {
+        String fileName = "all.cypher";
+        String query = "MATCH (start:Foo)-[rel:KNOWS]->(end:Bar) RETURN rel";
+        final Map<String, Object> config = withoutOptimization( map("format", "neo4j-shell") );
+
+        String exportQuery = "CALL apoc.export.cypher.query($query, $file, $config)";
+        Map<String, Object> params = Map.of("query", query, "config", config);
+
+        commonDataAndQueryAssertions(fileName, exportQuery, params,
+                EXPECTED_REL_ONLY, 0L, 1L, 1L);
+
+        final String exportData = """
+                MATCH (start:Foo)-[rel:KNOWS]->(end:Bar)
+                CALL apoc.export.cypher.data([], [rel], $file, $config)
+                YIELD nodes, relationships, properties RETURN *""";
+        commonDataAndQueryAssertions(fileName, exportData, params,
+                EXPECTED_REL_ONLY, 0L, 1L, 1L);
+
+
+        // check that `nodesOfRelationships: true` doesn't change the result
+        config.put("nodesOfRelationships", true);
+
+        commonDataAndQueryAssertions(fileName, exportQuery, params,
+                EXPECTED_REL_ONLY, 0L, 1L, 1L);
+
+        commonDataAndQueryAssertions(fileName, exportData, params,
+                EXPECTED_REL_ONLY, 0L, 1L, 1L);
+    }
```

Comment:
I'd suggest splitting this in another test. I'd advise to keep an assertion (or a group of assertions related to the same input) per tests. It's never good when a test fails and you have to dig out which input it failed for.

Reply:
Changed